docs/rfcs/batching-process-design.md

```diff
@@ -0,0 +1,282 @@
+interfaces an exporter component provides:
+
+- Merge(): when request batches have no upper bound. In this case, the interface
+  produces single outputs.
+- MergeSplit(): when there is a maximum size imposed. In this case, the interface
+  produces potentially more than one output request.
+
+Concurrency behavior varies. In the case where `MergeSplit()` is used, there is
+a potential for multiple requests to emit from a single request. In this case,
+steps 4 and 5 are executed repeatedly while there are more requests, meaning:
+
+1. Exports are synchronous and sequential.
+2. An error causes aborting of subsequent parts of the request.
+
+### Queue sender logic: existing
+
+The queue sender provides key functionality that determines the overall behavior
+of both batching components. When enabled, the queue sender will return success
+to the caller as soon as the request is enqueued. In the background, it concurrently
+exports requests in the queue using a configurable number of execution threads.
+
+It is worth evaluating the behavior of the queue sender with a persistent queue
+and with an in-memory queue:
+
+- Persistent queue: In this case, the queue stores the request before returning
+  success. This component is not directly responsible for data loss.
+- In-memory queue: In this case, the queue acts as a form of error suppression.
+  Callers do not wait for the export to return, so there is a chance of data loss
+  in this configuration.
+
+The queue sender does not consider the caller's context deadline when it attempts
+to enqueue the request. If the queue is full, the queue sender returns a queue-full
+error immediately.
+
+### Feature matrix
+
+| Support area | Batch processor | Batch sender | Explanation |
+| -----| -- |  -- | -- |
+| Merges requests | Yes | Yes | Does it merge smaller into larger batches? |
+| Splits requests | Yes | Yes, however sequential | Does it split larger into smaller batches? |
+| Cancellation  | No | No | Does it respect caller cancellation? |
+| Deadline  | No | No | Does it set an outgoing deadline? |
+| Metadata  | Yes | No | Can it batch by metadata key value(s)? |
+| Tracing  | No | No | Instrumented for tracing? |
+| Error transmission | No | Yes | Are export errors returned to callers? |
+| Concurrency enabled | No | Merge: Yes<br> MergeSplit: No | Does the component allow concurrent export? |
+| Independence | Yes | No | Are all data exported independently? |
+
+### Change proposal
+
+#### Queue sender defaults
+
+A central aspect of this presentation focuses on the queue sender,
+which along with one of the batching processors defines the behavior
+of most OpenTelemetry pipelines. Therefore, the default queue sender
+behavior is critical.
+
+This proposal argues in favor of the user, who does not want default
+behavior that leads to data loss. This requires one or more changes
+in the exporterhelper:
+
+1. Disable the queue sender by default. In this configuration, requests
```

Comment:
This is a significant change in the behavior. I don't remember any other user requests to make the in-memory queue disabled by default. I believe a significant portion of users would favor keeping in-memory queue enabled by default, even if it's prone to data loss in case of Collector's crash. If we want to change that, there should be a more extended discussion and evaluation. AFAIR, in-memory buffering is a common default setting in other data processing agents.